src/main/java/com/artipie/asto/s3/S3Storage.java

```diff
@@ -173,22 +157,16 @@ public S3Storage(final S3AsyncClient client, final String bucket, final boolean
         final CompletableFuture<Void> result;
         final Content onetime = new Content.OneTime(content);
         if (this.multipart) {
-            result = complementWithSize(onetime, S3Storage.MIN_MULTIPART).thenCompose(
-                updated -> {
-                    final CompletableFuture<Void> future;
-                    final Optional<Long> size = updated.size();
-                    if (size.isPresent() && size.get() < S3Storage.MIN_MULTIPART) {
-                        future = this.put(key, updated);
-                    } else {
-                        future = this.uploadMultipart(key, updated);
-                    }
-                    return future;
-                }
-            );
+            result =
+                new EstimatedContentCompliment(onetime, S3Storage.MIN_MULTIPART)
+                    .estimate()
+                    .toCompletableFuture()
+                    .thenCompose(estimated -> this.putMultipart(estimated, key));
```

Comment:
@andreoss we have lost the part of logic here: previosly, S3 storage used simple `put` method for content with a size less than `S3Storage.MIN_MULTIPART` and `uploadMultipart` only for content greater or equal to this value. Now each upload request goes to `putMultipart` method, and it's not correct, since it's more expencive for small data uploads (it requires 3 HTTP requests: 1) create multipart upload, get upload ID 2) upload all parts to this ID 3) finish upload by ID; where simple upload is just a one HTTP call)

Reply:
@g4s8 Done.